### DISCUSSION TITLE
[Art competition] Weak Update

### DISCUSSION THREAD

**Discussion by f52985** [prosyslab-classroom/cs524-program-analysis#272] [Art competition]
![DALL·E 2022-11-25 15 29 38 - A 3d render of a hero with a sword fighting against a dragon with multiple heads that spits fire](https://user-images.githubusercontent.com/26792793/203916478-4ae15e9f-bc76-47ff-95e5-1c7b7cf76836.png)

## Short Description
For me, "weak update" means "A 3d render of a hero with a sword fighting against a dragon with multiple heads that spits fire".

## Long Description (up to 3 sentences)
When fighting against a dragon with multiple heads, cutting one head is not enough to kill the main dragon, since there are still other heads that keep the dragon alive. This is analogous to a weak update. When updating an abstract location that can have multiple concrete locations, the updating should not fully overwrite (or kill) the original mapped value, since there are still other potential concrete locations that are not affected by the single actual concrete location update.

**Comment by KihongHeo** [comment]
Sounds like Cerberus
